### PR TITLE
api.md: document new ACL validate mode

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#4932

### CODE REVIEW

File: api.md [hunk 1]
```diff
@@ -16,15 +16,15 @@ Currently based on {some authentication method}. Visit the [admin panel](https:/
   - Authorize machine
     - [POST device authorized](#device-authorized-post)
   - Tags
-    - [POST device tags](#device-tags-post)
+    - [POST device tags](#device-tags-post)dfdf
```
Comment (maisem) [inline api.md:19]: debug?

Comment (bradfitz) [reply api.md:19]: LIVE PREVIEW WOKRED

File: api.md [hunk 2]
```diff
@@ -658,6 +680,8 @@ A 400 http status code and the errors in the response body.
 }
 ```
 
+An empty body or a JSON object with no `message` is returned on success.
```
Comment (maisem) [inline api.md:683]: It would be nice to document which of the two it is based on the input

**Review by maisem — APPROVED** [review]

File: api.md [hunk 3]
```diff
@@ -641,11 +646,28 @@ curl 'https://api.tailscale.com/api/v2/tailnet/example.com/acl/validate' \
...
+  -u "tskey-yourapikey123:" \
+  --data-binary '
+  {
+    "ACLs": [
+     { "Action": "accept", "src": ["100.105.106.107"], "dst": ["1.2.3.4:*"] },
+    ],
+    "Tests", [
+      {"src": "100.105.106.107", "allow": ["1.2.3.4:80"]}
+    ],
+  }
```
Comment (ramyfication) [inline api.md:662]:
```suggestion
  }'
```

File: api.md [hunk 4]
```diff
@@ -641,11 +646,28 @@ curl 'https://api.tailscale.com/api/v2/tailnet/example.com/acl/validate' \
...
+```
+
 Response:
-If all the tests pass, the response will be empty, with an http status code of 200.
 
-Failed test error response:
-A 400 http status code and the errors in the response body.  
+The HTTP status code will be 200 if the request was well formed and there were no server errors, even in the case of failing tests or an invalid ACL. Look at the response body to determine whether there was a problem withn your ACL or tests.
```
Comment (ramyfication) [inline api.md:667]:
```suggestion
The HTTP status code will be 200 if the request was well formed and there were no server errors, even in the case of failing tests or an invalid ACL. Look at the response body to determine whether there was a problem within your ACL or tests.
```

File: api.md (hunk 2 above)
Comment (bradfitz) [reply api.md:683]: Per the TODO in the API code, I want to change it later to return an object. For now I want to make people tolerant of either way.

File: api.md (hunk 3 above)
Comment (bradfitz) [reply api.md:662]: good eye

**Review by ramyfication — APPROVED** [review]

**Pull request merged**